plugins/org.jkiss.dbeaver.ext.snowflake.core/src/org/jkiss/dbeaver/ext/snowflake/model/SnowflakeSQLDialect.java

```diff
@@ -142,4 +142,32 @@ public String getColumnTypeModifiers(
         }
         return super.getColumnTypeModifiers(dataSource, column, typeName, dataKind);
     }
+
+    @Override
+    public boolean mustBeQuoted(@NotNull String str, boolean forceCaseSensitive) {
+        // Unquoted object identifiers:
+        // * Start with a letter (A-Z, a-z) or an underscore (“_”).
+        // * Contain only letters, underscores, decimal digits (0-9), and dollar signs (“$”).
+        // * Are stored and resolved as uppercase characters (e.g. id is stored and resolved as ID).
+        // https://docs.snowflake.com/en/sql-reference/identifiers-syntax
+
+        if (forceCaseSensitive || str.isBlank()) {
+            return true;
+        }
+        int firstCodePoint = str.codePointAt(0);
+        if (!isLatinLetter(firstCodePoint) && firstCodePoint != '_') {
+            return true;
+        }
+        for (int i = 1; i < str.codePointCount(0, str.length()); i++) {
```

Comment:
Please variable for `str.codePointCount(0, str.length())` otherwise it calculates it for every loop.